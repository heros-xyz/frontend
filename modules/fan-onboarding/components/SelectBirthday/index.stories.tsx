import { Box } from "@chakra-ui/react";
import type { Meta, StoryObj } from "@storybook/react";
import SelectBirthday from "./";

const meta: Meta<typeof SelectBirthday> = {
  title: "Fan/On Boarding/SelectBirthday",
  component: SelectBirthday,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;
type Story = StoryObj<typeof SelectBirthday>;

export const SelectBirthdayComponent: Story = {
  render: (args) => (
    <Box h="100vh" bg="secondary">
      <SelectBirthday {...args} />
    </Box>
  ),
};
