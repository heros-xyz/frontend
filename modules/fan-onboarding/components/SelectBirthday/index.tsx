import { Box, Flex, Text } from "@chakra-ui/react";
import * as Yup from "yup";
import { useFormik } from "formik";
import { ArrowRight } from "@/components/svg/ArrowRight";
import { BirthdateFanOnBoarding } from "@/components/svg/BirthdateFanOnBoarding";
import DateSelect from "@/components/ui/DateSelect";
import ErrorMessage from "@/components/common/ErrorMessage";
import { isValidDate } from "@/utils/functions";
import HerosOnboardingWrapperNew from "@/components/ui/HerosOnboardingWrapperNew";

interface initialValues {
  dateOfBirth: string;
}
interface IProp {
  onSubmit: (values: string) => void;
  initialValues: initialValues;
}

const EnterBirthday: React.FC<IProp> = ({ initialValues, onSubmit }) => {
  const validationSchema = Yup.object().shape({
    dateOfBirth: Yup.string()
      .required("This is a required field")
      .test("valid-date", "Invalid date", (value) => {
        return isValidDate(value);
      }),
  });

  const formik = useFormik({
    initialValues,
    validationSchema,
    onSubmit: (values) => {
      onSubmit(values.dateOfBirth);
    },
  });

  return (
    <HerosOnboardingWrapperNew
      Icon={
        <BirthdateFanOnBoarding
          w={{ base: "67px", xl: "107px" }}
          h={{ base: "80px", xl: "128px" }}
          color={{ base: "#FFFAE8", xl: "accent.4" }}
        />
      }
      textButton="Proceed"
      IconButton={<ArrowRight />}
      onSubmit={formik.handleSubmit}
      bgIconColor="accent.4"
      display="flex"
      alignItems="center"
      isPaddingTop={false}
    >
      <Box mb={{ base: 4, lg: 20 }} color="primary">
        <Box fontSize={{ lg: "xl" }} fontWeight="500" mb={3}>
          Select Date of Birth
          <Text as="span" color="error.dark">
            {" "}
            *
          </Text>
        </Box>
        <Flex
          fontSize={{ base: "sm", lg: "3xl" }}
          justifyContent="space-between"
          gap={{ base: 5, lg: 8 }}
        >
          <DateSelect
            date={initialValues?.dateOfBirth}
            onChange={(value) => formik.setFieldValue("dateOfBirth", value)}
            submitted={!!formik.submitCount}
            zIndex={20}
            isDarkTheme
          />
        </Flex>
        <ErrorMessage
          mt={0.5}
          condition={Boolean(formik.submitCount && formik.errors?.dateOfBirth)}
          errorMessage={formik.errors?.dateOfBirth}
        />
      </Box>
    </HerosOnboardingWrapperNew>
  );
};

export default EnterBirthday;
