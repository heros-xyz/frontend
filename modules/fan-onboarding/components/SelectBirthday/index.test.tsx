import { render } from "@testing-library/react";
import SelectBirthday from ".";

it("renders homepage unchanged", () => {
  const onChangeStep = (step: number) => {};
  const { container } = render(
    <SelectBirthday
      initialValues={{ dateOfBirth: "10/10/2012" }}
      onSubmit={() => onChangeStep(3)}
    />
  );
  expect(container).toMatchSnapshot();
});
