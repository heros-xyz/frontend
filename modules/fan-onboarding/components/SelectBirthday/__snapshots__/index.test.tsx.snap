// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="css-1g2wpaw"
  >
    <div
      class="css-1kmmdcq"
    />
    <div
      class="chakra-container css-p5v03p"
    >
      <div
        class="css-1ispe57"
      >
        <div
          class="css-cgkfdy"
        >
          <div
            class="css-1llc4z1"
          >
            <svg
              class="chakra-icon css-1pmed7j"
              focusable="false"
              viewBox="0 0 67 80"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_5448_58455)"
              >
                <path
                  d="M66.3644 32.7272C64.4553 28.9999 19.1826 30.3635 2.90987 37.5454C-2.99922 40.1817 0.455325 78.4544 10.4553 79.6363C18.2735 80.5454 35.0917 79.3635 42.0917 77.9999C44.9099 79.909 54.0917 76.0908 54.0008 71.5454C53.9099 68.9999 52.4553 66.909 51.0917 65.5454C52.4553 64.3635 53.4553 62.9999 53.7281 61.5454C54.1826 58.6363 52.0917 56.2726 50.3644 54.909C52.4553 53.6363 54.0917 51.8181 54.4553 49.909C54.9099 47.2726 52.6371 44.4544 50.5462 42.5454C63.6371 40.909 68.4553 36.6363 66.3644 32.7272Z"
                  fill="currentColor"
                />
                <path
                  d="M52.0012 74.5442C52.0012 74.5442 43.4557 80.8169 41.4557 77.2715C39.0921 73.0897 43.3648 70.2715 43.3648 70.2715C43.3648 70.2715 37.3648 72.3624 36.0921 68.726C34.8193 65.0897 35.4557 62.3624 41.8193 59.6351C41.8193 59.6351 34.0012 62.3624 32.4557 58.4533C30.3648 53.0897 36.6375 48.6351 43.4557 45.2715"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M26.0004 34.5454C31.0004 31.909 34.7276 37.8181 30.4549 42.2727C26.0913 46.6363 18.7276 47.8181 18.7276 47.8181C18.7276 47.8181 30.4549 56.9999 18.6367 60.6363"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M47.0921 32.2723C46.5466 32.2723 46.183 31.9086 46.0921 31.3632C46.0012 29.1814 46.4557 25.6359 46.4557 25.545C46.5466 24.9995 47.0012 24.6359 47.5466 24.7268C48.0921 24.8177 48.4557 25.2723 48.3648 25.8177C48.3648 25.8177 47.9103 29.2723 48.0012 31.2723C48.0921 31.7268 47.7285 32.1814 47.0921 32.2723Z"
                  fill="#FD2D55"
                />
                <path
                  d="M41.1819 32.4539C40.6365 32.4539 40.2729 32.0903 40.1819 31.5448C40.091 29.363 40.5456 25.8175 40.5456 25.7266C40.6365 25.1812 41.091 24.8175 41.6365 24.9085C42.1819 24.9994 42.5456 25.4539 42.4547 25.9994C42.4547 25.9994 42.0001 29.4539 42.091 31.4539C42.1819 31.9994 41.8183 32.4539 41.1819 32.4539C41.2729 32.4539 41.1819 32.4539 41.1819 32.4539Z"
                  fill="#FD2D55"
                />
                <path
                  d="M51.7275 24.7266C53.7275 22.8175 61.5457 14.8175 57.3639 15.272C56.7275 15.3629 55.273 16.3629 55.273 16.3629C55.273 16.3629 56.6366 13.7266 55.273 13.1811C54.0003 12.6357 51.8184 14.9084 51.8184 14.8175C51.7275 14.6357 53.3639 12.8175 52.5457 12.0902C52.1821 11.7266 50.7275 11.7266 48.5457 14.272C47.7275 15.272 50.9093 5.45384 49.3639 2.63566C47.4548 -0.818888 40.3639 -1.63707 37.273 4.90838C34.1821 11.4538 31.4548 22.4538 39.7275 26.0902C45.6366 28.6357 49.7275 26.7266 51.7275 24.7266Z"
                  fill="#FFA757"
                />
                <path
                  d="M38.3638 5.08984L32.9093 6.45348H32.8184L36.7275 9.99893L38.3638 5.08984Z"
                  fill="#FD2D55"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_5448_58455"
                >
                  <rect
                    fill="white"
                    height="80"
                    width="66.8182"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
        </div>
        <div
          class="css-14s5ckc"
        >
          <div
            class="css-1s2o82u"
          >
            <div
              class="css-1uoge8t"
            >
              <div
                class="css-coo02u"
              >
                <div
                  class="css-johq1t"
                >
                  Select Date of Birth
                  <span
                    class="chakra-text css-1b8yx6g"
                  >
                     
                    *
                  </span>
                </div>
                <div
                  class="css-1mavqzz"
                >
                  <div
                    class="css-17eahu8"
                  >
                    <div
                      class=" css-n9lnso-SelectContainer"
                    >
                      <span
                        class="css-1f43avz-a11yText-A11yText"
                        id="react-select-:r0:-live-region"
                      />
                      <span
                        aria-atomic="false"
                        aria-live="polite"
                        aria-relevant="additions text"
                        class="css-1f43avz-a11yText-A11yText"
                      />
                      <div
                        class=" css-1xuao1n-control"
                      >
                        <div
                          class=" css-15ki4p9-ValueContainer"
                        >
                          <div
                            class="css-xxdqwu"
                          >
                            <div
                              class="css-1gmy4hn"
                            >
                              10
                            </div>
                            <div
                              class=" css-1fs6gt4-Input"
                              data-value=""
                            >
                              <input
                                aria-autocomplete="list"
                                aria-expanded="false"
                                aria-haspopup="true"
                                autocapitalize="none"
                                autocomplete="off"
                                autocorrect="off"
                                class=""
                                id="react-select-:r0:-input"
                                role="combobox"
                                spellcheck="false"
                                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                                tabindex="0"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class=" css-1hb7zxy-IndicatorsContainer"
                        >
                          <span
                            class=" css-1xjgjl1-IndicatorSeparator"
                          />
                          <svg
                            class="chakra-icon css-h50qoc"
                            focusable="false"
                            viewBox="0 0 20 20"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M16.25 6.875L10 13.125L3.75 6.875"
                              stroke="currentColor"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="1.5"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="css-17eahu8"
                  >
                    <div
                      class=" css-n9lnso-SelectContainer"
                    >
                      <span
                        class="css-1f43avz-a11yText-A11yText"
                        id="react-select-:r1:-live-region"
                      />
                      <span
                        aria-atomic="false"
                        aria-live="polite"
                        aria-relevant="additions text"
                        class="css-1f43avz-a11yText-A11yText"
                      />
                      <div
                        class=" css-1xuao1n-control"
                      >
                        <div
                          class=" css-15ki4p9-ValueContainer"
                        >
                          <div
                            class="css-xxdqwu"
                          >
                            <div
                              class="css-1gmy4hn"
                            >
                              Oct
                            </div>
                            <div
                              class=" css-1fs6gt4-Input"
                              data-value=""
                            >
                              <input
                                aria-autocomplete="list"
                                aria-expanded="false"
                                aria-haspopup="true"
                                autocapitalize="none"
                                autocomplete="off"
                                autocorrect="off"
                                class=""
                                id="react-select-:r1:-input"
                                role="combobox"
                                spellcheck="false"
                                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                                tabindex="0"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class=" css-1hb7zxy-IndicatorsContainer"
                        >
                          <span
                            class=" css-1xjgjl1-IndicatorSeparator"
                          />
                          <svg
                            class="chakra-icon css-h50qoc"
                            focusable="false"
                            viewBox="0 0 20 20"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M16.25 6.875L10 13.125L3.75 6.875"
                              stroke="currentColor"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="1.5"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="css-17eahu8"
                  >
                    <div
                      class=" css-n9lnso-SelectContainer"
                    >
                      <span
                        class="css-1f43avz-a11yText-A11yText"
                        id="react-select-:r2:-live-region"
                      />
                      <span
                        aria-atomic="false"
                        aria-live="polite"
                        aria-relevant="additions text"
                        class="css-1f43avz-a11yText-A11yText"
                      />
                      <div
                        class=" css-1xuao1n-control"
                      >
                        <div
                          class=" css-15ki4p9-ValueContainer"
                        >
                          <div
                            class="css-xxdqwu"
                          >
                            <div
                              class="css-1gmy4hn"
                            >
                              2012
                            </div>
                            <div
                              class=" css-1fs6gt4-Input"
                              data-value=""
                            >
                              <input
                                aria-autocomplete="list"
                                aria-expanded="false"
                                aria-haspopup="true"
                                autocapitalize="none"
                                autocomplete="off"
                                autocorrect="off"
                                class=""
                                id="react-select-:r2:-input"
                                role="combobox"
                                spellcheck="false"
                                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                                tabindex="0"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class=" css-1hb7zxy-IndicatorsContainer"
                        >
                          <span
                            class=" css-1xjgjl1-IndicatorSeparator"
                          />
                          <svg
                            class="chakra-icon css-h50qoc"
                            focusable="false"
                            viewBox="0 0 20 20"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M16.25 6.875L10 13.125L3.75 6.875"
                              stroke="currentColor"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="1.5"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="css-lrby83"
              >
                <button
                  class="chakra-button css-k6pd2s"
                  type="button"
                >
                  Proceed
                  <span
                    class="css-9n3bbc"
                  >
                    <svg
                      class="chakra-icon css-iz364c"
                      focusable="false"
                      viewBox="0 0 15 14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M1.25 7L13.75 7M13.75 7L8.125 1.375M13.75 7L8.125 12.625"
                        stroke="#33EFEF"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="1.5"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
          <button
            class="chakra-button css-1mv50kv"
            type="button"
          >
            Proceed
            <span
              class="css-9n3bbc"
            >
              <svg
                class="chakra-icon css-iz364c"
                focusable="false"
                viewBox="0 0 15 14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1.25 7L13.75 7M13.75 7L8.125 1.375M13.75 7L8.125 12.625"
                  stroke="#33EFEF"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
